src/tasks/netSdk/NetSdkTaskHelper.ts

```diff
@@ -0,0 +1,138 @@
+import { DockerTaskExecutionContext, addVolumeWithoutConflicts, getDefaultContainerName, getDefaultImageName, getMounts } from "../TaskHelper";
+import { NetCoreTaskHelper } from "../netcore/NetCoreTaskHelper";
+
+/**
+ * Native architecture of the current machine in the RID format
+ * {@link https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/runtime.json}
+ */
+export type RidCpuArchitecture =
+    | 'x64'
+    | 'x86'
+    | 'arm64'
+    | 'arm'
+    | 'ppc64le'
+    | 'mips64'
+    | 's390x'
+    | string;
+
+export const NET_SDK_RUN_TASK_SYMBOL = 'dotnet-container-sdk';
+const IMAGE_TAG = 'dev'; // intentionally default to dev tag for phase 1 of this feature
+const ERR_MSG_NO_WORKSPACE_FOLDER = l10n.t(`Unable to determine task scope to execute task ${NET_SDK_RUN_TASK_SYMBOL}. Please open a workspace folder.`);
+
+export class NetSdkTaskHelper {
+
+    public async getNetSdkBuildCommand(context: DockerTaskExecutionContext): Promise<string> {
+
+        const configuration = 'Debug'; // intentionally default to Debug configuration for phase 1 of this feature
+        const projPath = await this.inferProjPath(context.actionContext, context.folder);
+
+        // {@link https://github.com/dotnet/sdk-container-builds/issues/141} this could change in the future
+        const publishFlag = NetCoreTaskHelper.isWebApp(projPath) ? '-p:PublishProfile=DefaultContainer' : '/t:PublishContainer';
+
+        const folderName = await quickPickWorkspaceFolder(context.actionContext, ERR_MSG_NO_WORKSPACE_FOLDER);
+
+        const args = composeArgs(
+            withArg('dotnet', 'publish'),
+            withNamedArg('--os', await this.normalizeOsToRid()),
+            withNamedArg('--arch', await this.normalizeArchitectureToRid()),
+            withArg(publishFlag),
+            withNamedArg('--configuration', configuration),
+            withNamedArg('-p:ContainerImageName', getValidImageName(folderName.name), { assignValue: true }),
+            withNamedArg('-p:ContainerImageTag', IMAGE_TAG, { assignValue: true })
+        )();
+
+        const quotedArgs = Shell.getShellOrDefault().quote(args);
+        return quotedArgs.join(' ');
+    }
+
+    public async getNetSdkRunCommand(context: DockerTaskExecutionContext): Promise<string> {
+        const client = await ext.runtimeManager.getClient();
+        const folderName = await quickPickWorkspaceFolder(context.actionContext, ERR_MSG_NO_WORKSPACE_FOLDER);
+
+        const command = await client.runContainer({
+            detached: true,
+            publishAllPorts: true,
+            name: getDefaultContainerName(folderName.name, IMAGE_TAG),
+            environmentVariables: {},
+            removeOnExit: true,
+            imageRef: getDefaultImageName(folderName.name, IMAGE_TAG),
+            labels: defaultVsCodeLabels,
+            mounts: await this.getRemoteDebuggerMount(),
+            customOptions: '--expose 8080',
```

Comment:
We should add a new `expose` property to `RunContainerCommandOptions` rather than falling back to customOptions. Ideally you could specify `expose: [8080],` here.